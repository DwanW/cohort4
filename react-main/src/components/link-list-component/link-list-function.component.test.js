import React from 'react';
import { shallow } from 'enzyme';
import ListApp from './link-list-function.component';

describe('LinkedList component test', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ListApp />);
    });

    it('should render linked list component correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render initial listNode correctly', () => {
        let nodes = wrapper.find(".list-node-container").children();
        expect(nodes.length).toBe(3);
        expect(nodes.at(0).text()).toBe('apple 12');
        expect(nodes.at(1).text()).toBe('orange 100');
        expect(nodes.at(2).text()).toBe('banana 75');
    });
});