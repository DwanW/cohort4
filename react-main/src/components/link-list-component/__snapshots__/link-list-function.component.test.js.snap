// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinkedList component test should render linked list component correctly 1`] = `ShallowWrapper {}`;
